core/domain/state_domain_test.py

```diff
@@ -1329,14 +1341,16 @@ def test_cannot_update_interaction_solution_with_non_dict_solution(self):
         }
 
         exploration.init_state.update_interaction_hints(hints_list)
-        exploration.init_state.update_interaction_solution(solution)
+        exploration.init_state.update_interaction_solution(
+            state_domain.Solution.from_dict(
+                exploration.init_state.interaction.id, solution))
 
         self.assertEqual(
             exploration.init_state.interaction.solution.to_dict(), solution)
 
         with self.assertRaisesRegexp(
-            Exception, 'Expected solution to be a dict'):
-            exploration.init_state.update_interaction_solution([])
+            Exception, 'list indices must be integers, not unicode'):
```

Comment:
This is an unexpected error, isn't it?

Reply:
sorry,I didn't get you.can you elaborate?

Reply:
@DubeySandeep PTAL , i didn't get your comment.

Reply:
HI @anmolshkl, sorry for the misunderstanding!

My question:
Are we expecting to get this exception from `update_interaction_solution` method?

You have resolved this comment in the latest commit!

Reply:
yes I have made the required changes in this section.
PTAL at the latest commit.